.gitlab-ci.yml

```diff
@@ -55,6 +55,17 @@ create_key:
 
 # STATIC ANALYSIS
 
+static-analysis:
+  stage: analysis
+  variables:
+    DETEKT_PUBLIC_API: "true"
+    DETEKT_CUSTOM_RULES_BUILD_TASK: ":tools:detekt:jar"
+    DETEKT_CUSTOM_RULES_JAR_PATH: "tools/detekt/build/libs/detekt.jar"
+    DETEKT_CUSTOM_RULES_YML_PATH: "detekt_custom.yml"
```

Comment:
this goes surely too deep for now but what do you think about just pointing to a say `datadog-static-analysis-detekt-config.{yml/json...}` (couldnt care less about the format) where all these variables would be defined; giving more freedom to each project (and bonus 
* avoid an ever grwoing list of env variable 
* file versioned in each project (so less "magic"; somebody could rename the task on one project not being aware it is used by `dd-source`)

Reply:
Yeah that makes sense, but indeed maybe this is going too deep for now ;)